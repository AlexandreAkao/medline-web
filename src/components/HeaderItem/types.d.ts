interface IHeaderItemProps extends IButtonProps {
  as: 'button' | 'link';
  href?: string;
  children: React.ReactNode;
}
