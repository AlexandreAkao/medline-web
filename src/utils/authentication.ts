import { isEmpty } from 'ramda';

export const getUserInfo = () => {
  const user = JSON.parse(localStorage.getItem('user') ?? '{}');

  return isEmpty(user) ? null : (user as IUser);
};

export const geToken = () => localStorage.getItem('token');
