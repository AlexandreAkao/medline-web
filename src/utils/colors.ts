export const getColorAlpha = (color: string, alpha = 1) => {
  const alphaLevel = Math.round(255 * alpha).toString(16);
  return color + alphaLevel;
};
