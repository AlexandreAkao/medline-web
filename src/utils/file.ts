export const downloadFile = (blob: Blob, filename: string) => {
  const link = document.createElement('a');
  link.href = URL.createObjectURL(blob);
  link.download = filename;
  link.click();
};
