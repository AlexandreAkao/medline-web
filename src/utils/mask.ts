export enum Mask {
  Empyt = '',
  CPF = '999.999.999-99',
  Date = '99/99/9999',
}
