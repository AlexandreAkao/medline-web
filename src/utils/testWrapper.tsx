import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

export const renderWithRouter = (component: React.ReactNode) =>
  render(<MemoryRouter>{component}</MemoryRouter>);
