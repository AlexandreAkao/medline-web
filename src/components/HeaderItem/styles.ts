import styled from 'styled-components';

import colors from 'styles/colors';

export const HeaderItemContainer = styled.div`
  a {
    color: ${colors.white.normal};
    font-weight: 700;
    font-size: 1rem;
    transition: 0.2s ease all;
    padding: 10px;
    display: flex;
    justify-content: center;
    align-items: center;

    :hover {
      transition: 0.2s ease all;
      color: ${colors.white.light};
    }

    * {
      margin-left: 8px;
    }
  }
`;
