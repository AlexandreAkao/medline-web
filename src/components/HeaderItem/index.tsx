import { useMemo } from 'react';
import { Link } from 'react-router-dom';

import Button from 'components/Button';
import { HeaderItemContainer } from 'components/HeaderItem/styles';

function HeaderItem({ as = 'link', children, href = '', ...buttonProps }: IHeaderItemProps) {
  const Item = useMemo(() => {
    const itemsMap = {
      button: <Button {...buttonProps}>{children}</Button>,
      link: <Link to={href}>{children}</Link>,
    };
    return itemsMap[as];
  }, [as, buttonProps, children, href]);

  return <HeaderItemContainer>{Item}</HeaderItemContainer>;
}

export default HeaderItem;
