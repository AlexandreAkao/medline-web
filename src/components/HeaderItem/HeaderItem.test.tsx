import { screen } from '@testing-library/react';

import HeaderItem from 'components/HeaderItem';
import { renderWithRouter } from 'utils/testWrapper';

describe('HeaderItem', () => {
  const renderComponent = (customProps: Partial<IHeaderItemProps> = {}) => {
    const props: IHeaderItemProps = {
      as: 'button',
      children: undefined,
      ...customProps,
    };
    return renderWithRouter(<HeaderItem {...props} />);
  };

  it('should match with snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render as button', () => {
    renderComponent({ as: 'button' });
    const headerItem = screen.getByRole('button');
    expect(headerItem).toBeInTheDocument();
  });

  it('should render as link', () => {
    renderComponent({ as: 'link', href: '/' });
    const headerItem = screen.getByRole('link');
    expect(headerItem).toBeInTheDocument();
  });
});
