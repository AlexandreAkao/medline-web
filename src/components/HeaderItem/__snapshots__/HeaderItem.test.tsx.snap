// Vitest Snapshot v1

exports[`HeaderItem > should match with snapshot 1`] = `
<div>
  <div
    class="sc-gsnTZi qZpGk"
  >
    <button
      class="sc-bczRLJ bJBahq"
      type="button"
    />
  </div>
</div>
`;
